src/utils/OnboardingProvider.tsx

```diff
@@ -0,0 +1,293 @@
+'use client';
+
+import supabase from '@/api/supabase/createClient';
+import {
+  deleteLanguages,
+  deleteRoles,
+  upsertLanguages,
+  upsertProfile,
+  upsertRoles,
+} from '@/api/supabase/queries/profiles';
+import { Profile, ProfileLanguage, ProfileRole } from '@/types/schema';
+import { UUID } from 'crypto';
+import {
+  createContext,
+  useState,
+  useMemo,
+  ReactNode,
+  SetStateAction,
+  Dispatch,
+  useEffect,
+} from 'react';
+
+interface FlowData {
+  url: string;
+  name: string;
+}
+
+interface OnboardingContextType {
+  profile: Profile;
+  languages: ProfileLanguage[];
+  roles: ProfileRole[];
+  progress: number;
+  flow: FlowData[];
+  canContinue: boolean;
+  flushProfileData: () => Promise<void>;
+  updateProfile: (updateInfo: Partial<Profile>) => Promise<void>;
+  setProgress: Dispatch<SetStateAction<number>>;
+  setCanContinue: Dispatch<SetStateAction<boolean>>;
+  setFlow: Dispatch<SetStateAction<FlowData[]>>;
+  setLanguages: (languages: ProfileLanguage[]) => Promise<void>;
+  setRoles: (roles: ProfileRole[]) => Promise<void>;
+}
+
+export const OnboardingContext = createContext<
+  OnboardingContextType | undefined
+>(undefined);
+
+const blankProfile: Profile = {
+  first_name: '',
+  last_name: '',
+  hours_per_month: 0,
+  location: '',
+  start_date: '',
+  user_id: '0-0-0-0-0',
+};
+
+export default function OnboardingProvider({
+  children,
+}: {
+  children: ReactNode;
+}) {
+  const [progress, setProgress] = useState(0);
+  const [flow, setFlow] = useState<FlowData[]>([]);
```

Comment:
to clarify, will page 0 ultimately be the role selection page?

Reply:
Yup!